src/index.js

```diff
@@ -1,5 +1,5 @@
-import template from 'babel-template';
```

Comment:
I'm not a maintainer of this module, but FYI this whole module could be rewritten as
```
export default ({ template }) => {
  const buildImport = template(`
    (new Promise((resolve) => {
      require.ensure([], (require) => {
        resolve(require(SOURCE));
      });
    }))
  `);
  
  return {
    manipulateOptions(opts, parserOpts) {
      parserOpts.plugins.push("dynamicImport");
    },

    visitor: {
      Import(path) {
        const newImport = buildImport({
          SOURCE: path.parentPath.node.arguments,
        });
        path.parentPath.replaceWith(newImport);
      },
    },
  };
};
```
and work on both Babel 6 and 7, I'd expect. Generally we don't recommend importing `babel-template` or `@babel/template` directly. The expectations are less clear for `@babel/plugin-syntax-dynamic-import`, but there's nothing wrong with just setting the parser plugin directly if you want.

Reply:
TIL. This seems like a much better alternative to releasing a breaking change that only supports Babel v7. I'll update this PR. Thanks @loganfsmyth!